flink-table-store-spark/src/test/java/org/apache/flink/table/store/spark/SparkReadITCase.java

```diff
@@ -281,6 +281,176 @@ public void testAddColumn() throws Exception {
+                .isEqualTo(
+                        "[[CREATE TABLE testRenameColumn (\n"
+                                + "  `aa` INT NOT NULL,\n"
+                                + "  `b` BIGINT,\n"
+                                + "  `c` STRING)\n"
+                                + "]]");
+    }
+
+    @Test
+    public void testRenamePartitionKey() {
+        spark.sql("USE tablestore");
+        spark.sql(
+                "CREATE TABLE default.testRenamePartitionKey (\n"
+                        + "a BIGINT,\n"
+                        + "b STRING) USING tablestore\n"
+                        + "COMMENT 'table comment'\n"
+                        + "PARTITIONED BY (a)\n"
+                        + "TBLPROPERTIES ('foo' = 'bar')");
+
+        List<Row> beforeRename =
+                spark.sql("SHOW CREATE TABLE tablestore.default.testRenamePartitionKey")
+                        .collectAsList();
+        assertThat(beforeRename.toString())
+                .isEqualTo(
+                        "[[CREATE TABLE testRenamePartitionKey (\n"
+                                + "  `a` BIGINT,\n"
+                                + "  `b` STRING)\n"
+                                + "PARTITIONED BY (a)\n"
+                                + "]]");
+
+        assertThatThrownBy(
+                        () ->
+                                spark.sql(
+                                        "ALTER TABLE tablestore.default.testRenamePartitionKey RENAME COLUMN a to aa"))
+                .isInstanceOf(RuntimeException.class)
+                .hasMessage("java.lang.UnsupportedOperationException: Cannot rename partition key");
+    }
+
+    @Test
+    public void testDropSingleColumn() throws Exception {
+        Path tablePath = new Path(warehousePath, "default.db/testDropSingleColumn");
+        createTestHelper(tablePath);
+
+        List<Row> beforeRename =
+                spark.sql("SHOW CREATE TABLE tablestore.default.testDropSingleColumn")
+                        .collectAsList();
+        assertThat(beforeRename.toString())
+                .isEqualTo(
+                        "[[CREATE TABLE testDropSingleColumn (\n"
+                                + "  `a` INT NOT NULL,\n"
+                                + "  `b` BIGINT,\n"
+                                + "  `c` STRING)\n"
+                                + "]]");
+
+        spark.sql("ALTER TABLE tablestore.default.testDropSingleColumn DROP COLUMN a");
+
+        List<Row> afterRename =
+                spark.sql("SHOW CREATE TABLE tablestore.default.testDropSingleColumn")
+                        .collectAsList();
+        assertThat(afterRename.toString())
+                .isEqualTo(
+                        "[[CREATE TABLE testDropSingleColumn (\n"
+                                + "  `b` BIGINT,\n"
+                                + "  `c` STRING)\n"
+                                + "]]");
+    }
+
+    @Test
+    public void testDropColumns() throws Exception {
```

Comment:
Test that table can still be read.

Reply:
Read data will fail in this PR. I found this issue [https://issues.apache.org/jira/browse/FLINK-27846](FLINK-27846) before I fix FLINK-27847. Should we fix read data in FLINK-27847 or in FLINK-27846? What do you think of it? HTX

Reply:
"Read data will fail in this PR." What causes this issue? Are there any differences between adding a column and renaming or dropping a column? If there are they should be fixed before merging this PR. That is, let's first implement FLINK-27846 and then come back to this PR.

Reply:
> 

Sounds good to me, I will fix [FLINK-27846](https://issues.apache.org/jira/browse/FLINK-27846) first

Reply:
It's exciting, if [FLINK-27846](https://issues.apache.org/jira/browse/FLINK-27846) can be fixed.